server/src/index.js

```diff
@@ -11,52 +11,83 @@ const url = process.env.URL
 app.listen(3000, () => console.log('Listening to port 3000'))
 app.get('/user/:id', (req, res) => res.send(`Welcome to the homepage of user ${req.params.id}.`))
 
+function createPost(req, type) {
```

Comment:
This is a bad separation of concern. This function duplicates the logic of `switch (type)`. Also `type` is in `req.body`, which is a duplicate parameter. Third, `req` is an Express request object. This function should have no knowledge of Express.

Maybe put this back for now. We can think about reducing complexity later.